src/store/sagas/sagaItems.js

```diff
@@ -1,27 +1,27 @@
 import { takeEvery, put, call, select } from 'redux-saga/effects';
 import { API_PHONE_INFO_REQUEST, API_PHONE_INFO_SUCCESS, } from '../types/types';
 
+import { apiDataFetch } from '../../utils/genericFunctions';
 import { fonoapiURL, fonoToken } from '../../utils/constants';
 
 function reqData(brandName) {
-    return fetch(`${fonoapiURL}/v1/getdevice?token=${fonoToken}&device=${brandName}`)
-        .then(res => res.json())
+   return apiDataFetch(fonoapiURL, fonoToken, brandName)
 }
 
 function* fetchData() {
-    const getBrand = yield select();
-    const brandName = getBrand.phonesData.brand;
+   const getBrand = yield select();
```

Comment:
https://redux-saga.js.org/docs/api/#selectselector-args
create a selector 
```js 
export const getBrandName = state => state.phonesData.brand
```
```suggestion
   const getBrand = yield select(getBrandName);
```